tests/test-include/fake-sys-include.h

```diff
@@ -0,0 +1,10 @@
+#define FAKE_SYS_SCALE_ONE 1
+
+// Note - files included this way cannot transitively include any files CUDA/ROCm won't compile
+// These are bad
+// #include <ceed.h>
+// #include <math.h>
+// #include <stddef.h>
```

Comment:
This is the sharp edge here and why we can't currently totally replace all our JiT infrastructure with this approach (yet?). A simple include of `ceed.h` which is in a lot of QFunction code brings in these lines
```
#include <stdarg.h>
#include <stdbool.h>
#include <stdio.h>
```
and CUDA/ROCm hate those includes.

So currently this PR keeps the chasing of the local includes via "" to build a string buffer but permits system includes via <> to add additional source dirs inside of the user QFunction source.

Reply:
I think we can merge this as is and make a second attack at this once we're ready to revamp SYCL (I just got my access to the machine but need to sign up and such)

Reply:
That makes sense. We could move the necessary declarations to `ceedtypes.h` and have qfunction headers only include those, not `ceed.h`.

Reply:
I think `ceed/types.h` has everything that could be used in a QFunction

Reply:
Maybe we should explicitly recommend that. No need to force people to change, but making the compilation environment (and context present in an IDE) as close as possible to the JIT environment would be a good thing for low-friction workflow.

Reply:
We could have an ifdef that swaps `ceed.h` to include `ceed/types.h` instead of `ceed/ceed.h` during JiT?

Reply:
Note to me:

1) use `code << #include <ceed/jit-source/cuda/cuda-jit.h>;`

2) could maybe `code << #include " << source_path;`

3) then could `-I << source_dir;` for above file?

Reply:
Ok, 1 was really easy to do and should save us some headache. 2 and 3 can probably wait for a follow-up?